sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/FunctionCallingSample.java

```diff
@@ -46,62 +48,65 @@ public static void main(String[] args) {
-        List<ChatRequestMessage> chatMessages2 = handleFunctionCallResponse(chatCompletions.getChoices(), chatMessages);
+        ChatCompletions chatCompletions = client.getChatCompletions(deploymentOrModelId,
+                new ChatCompletionsOptions(chatRequestMessages)
+                        .setFunctionCall(FunctionCallConfig.AUTO)
+                        .setFunctions(functions));
 
         // Take your function_call result as the input prompt to make another request to service.
-        ChatCompletionsOptions chatCompletionOptions2 = new ChatCompletionsOptions(chatMessages2);
-        ChatCompletions chatCompletions2 = client.getChatCompletions(deploymentOrModelId, chatCompletionOptions2);
-        List<ChatChoice> choices = chatCompletions2.getChoices();
-        ChatResponseMessage message = choices.get(0).getMessage();
-        System.out.printf("Message: %s.%n", message.getContent());
+        chatRequestMessages = handleFunctionCallResponse(chatCompletions.getChoices(),
+                chatRequestMessages);
+        ChatCompletions chatCompletionsAnswer = client.getChatCompletions(deploymentOrModelId,
+                new ChatCompletionsOptions(chatRequestMessages));
+
+        System.out.printf("Message: %s.%n", chatCompletionsAnswer.getChoices().get(0).getMessage().getContent());
     }
 
-    private static Map<String, Object> getFunctionDefinition() {
-        // Construct JSON in Map, or you can use create your own customized model.
-        Map<String, Object> location = new HashMap<>();
-        location.put("type", "string");
-        location.put("description", "The city and state, e.g. San Francisco, CA");
-        Map<String, Object> unit = new HashMap<>();
-        unit.put("type", "string");
-        unit.put("enum", Arrays.asList("celsius", "fahrenheit"));
-        Map<String, Object> prop1 = new HashMap<>();
-        prop1.put("location", location);
-        prop1.put("unit", unit);
-        Map<String, Object> functionDefinition = new HashMap<>();
-        functionDefinition.put("type", "object");
-        functionDefinition.put("required", Arrays.asList("location", "unit"));
-        functionDefinition.put("properties", prop1);
-        return functionDefinition;
+    private static FunctionParameters getCurrentWeatherFunctionParameters() {
+        FunctionProperties location = new FunctionProperties()
+                .setType("string")
+                .setDescription("The city and state, e.g. San Francisco, CA");
+
+        FunctionProperties format = new FunctionProperties()
+                .setType("string")
+                .setEnumString(Arrays.asList("celsius", "fahrenheit"))
+                .setDescription("The temperature unit to use. Infer this from the users location.");
+
+        Map<String, FunctionProperties> props = new HashMap<>();
+        props.put("location", location);
+        props.put("format", format);
```

Comment:
I think that you should rename `format` to `unit` as this is a bit confusing. This is for requesting the temperature in either Fahrenheit or Celsius, right?

Reply:
The `json` in the article describes the field as:

```json
"format": {
    "type": "string",
    "enum": ["celsius", "fahrenheit"],
    "description": "The temperature unit to use. Infer this from the users location.",
},
```

I think that maybe it's a mistake in the naming, I think we could fix it. Specially, because in our other samples we have similar mehtods and we call it `unit` instead of `format`. I would personally prefer that we align with our own samples in this regard.

Reply:
Updated